platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/DefaultProjectManager.java

```diff
@@ -578,26 +562,51 @@ public void updateProjectConfig(Project project, ProjectConfig config)
             throw new ServerException(e.getServiceError());
         }
 
-        updateProjectInWorkspace(project.getWorkspace(), projectConfig);
+        filterAttributesRecursive(projectConfigDto, types, project.getBaseFolder());
+
+        updateProjectInWorkspace(project.getWorkspace(), projectConfigDto);
     }
 
-    private void getModulesRecursive(ModuleConfig moduleConfig, List<ModuleConfigDto> projectModules) {
-        List<ModuleConfigDto> modules = new ArrayList<>();
+    private void filterAttributesRecursive(ProjectConfigDto projectConfig,
+                                           ProjectTypes projectTypes,
+                                           FolderEntry folderEntry) throws ValueStorageException,
+                                                                           ProjectTypeConstraintException,
+                                                                           ServerException,
+                                                                           ForbiddenException {
+        Map<String, List<String>> runtimeAttributes = new HashMap<>();
```

Comment:
Aa i can see `runtimeAttributes` are used only in line *580*, so if you don't need this attributes you should change the logic of `filterAttributes` or use `Collection.emptyMap()` which may also require `filterAttributes` method changes as it is immutable.